src/tools/imgtool/modules/hp85_tape.cpp

```diff
@@ -0,0 +1,1192 @@
+} dir_entry_85_t;
+
+/********************************************************************************
+ * Tape image
+ ********************************************************************************/
+class tape_image_85_t {
+public:
+	tape_image_85_t(void);
+
+	bool is_dirty(void) const { return dirty; }
+
+	void format_img(void);
+
+	imgtoolerr_t load_from_file(imgtool::stream *stream);
+	typedef std::vector<uint8_t> sif_file_t;
+	typedef std::unique_ptr<sif_file_t> sif_file_ptr_t;
+	bool load_sif_file(file_no_t file_no , sif_file_t& out);
+	imgtoolerr_t save_to_file(imgtool::stream *stream);
+
+	bool get_dir_entry(unsigned idx , const dir_entry_85_t*& entry) const;
+	bool find_file(const char *filename , bool ignore_ext , unsigned& idx) const;
+	bool alloc_new_file(dir_entry_85_t*& entry , sif_file_ptr_t&& file_data);
+	bool delete_dir_entry(unsigned idx);
+	bool finalize_allocation();
+	static void get_filename_and_ext(const dir_entry_85_t& ent , bool inc_ext , char *out);
+	static void split_filename_and_ext(const char *filename , char *fname , char *ext);
+
+private:
+	// Tape image
+	hti_format_t image;
+	bool dirty;
+	// Directory
+	std::vector<dir_entry_85_t> dir;
+	// Content
+	std::vector<sif_file_ptr_t> content;
+	// First file on track 1
+	file_no_t file_track_1;
+	// No. of first record on track 1
+	uint16_t record_track_1;
+
+	bool dec_rec_header(const tape_word_t *hdr , file_no_t& file_no , uint16_t& rec_no , bool& has_body , unsigned& body_len);
+	bool load_whole_tape();
+	static tape_word_t checksum(const tape_word_t *block , unsigned block_len);
+	bool decode_dir(const sif_file_t& file_0);
+	void encode_dir(sif_file_t& file_0) const;
+	void save_words(unsigned track , tape_pos_t& pos , const tape_word_t *block , unsigned block_len);
+	void save_sif_file(unsigned& track , tape_pos_t& pos , file_no_t file_no , const sif_file_t& in);
+	static bool filename_char_check(uint8_t c);
+	static bool filename_check(const uint8_t *filename);
+};
+
+/********************************************************************************
+ * Image state
+ ********************************************************************************/
+typedef struct {
+	imgtool::stream *stream;
+	tape_image_85_t *img;
+} tape_state_t;
+
+/********************************************************************************
+ * Directory enumeration
+ ********************************************************************************/
+typedef struct {
+	unsigned dir_idx;
+} dir_state_t;
+
+/********************************************************************************
+ * Internal functions
+ ********************************************************************************/
+tape_image_85_t::tape_image_85_t(void)
+	: dirty(false)
+{
+}
+
+void tape_image_85_t::format_img(void)
+{
+	// Create an empty directory
+	dir.clear();
+	content.clear();
+
+	// Allocate space
+	finalize_allocation();
+}
+
+static int my_seekproc(void *file, int64_t offset, int whence)
```

Comment:
Rather than putting `static` everywhere, could you please group things that don't need symbols exported in an anonymous namespace?  It reduces clutter.